### PR TITLE
Add: [Actions] Check CI annotations to detect compile warnings

### PR DESCRIPTION
**Pull request by glx22**
## Motivation / Problem
When compilation generates warnings in a PR, they are visible in `checks` tab and inlined in `Files changed` tab, but they are easy to miss if not actively looked for.

## Description
This adds a CI job to check if there are any annotations generated by previous jobs, and "fails" CI if any is found.
And I enabled this job for PRs only.

A test PR with compile warnings is available [here](https://github.com/glx22/OpenTTD/pull/13).

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, gs_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

**Comment by LordAro** [comment]
image
output of it is not ideal...

File: .github/workflows/ci-build.yml [hunk 1]
```diff
@@ -315,3 +315,41 @@ jobs:
...
+        route: GET /repos/{repository}/check-suites/{check_suite_id}/check-runs
+        repository: ${{ github.repository }}
+        check_suite_id: ${{ fromJson(steps.check_suite_id.outputs.data).check_suite_id }}
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Check annotations
+      shell: bash
+      run: |
+        exit $(echo '${{ toJson(fromJson(steps.check_runs.outputs.data).check_runs.*.output.annotations_count) }}' | jq 'reduce .[] as $i (0; . + $i)')
```
Comment (TrueBrain) [inline .github/workflows/ci-build.yml]: Given we have the JSON output here .. can't we do something fun with it, like reporting how many annotations there were? Simple embedded Python script of sorts?

PS: I really like what you did here, this is just bikeshedding :)

File: .github/workflows/ci-build.yml [hunk 2]
```diff
@@ -315,3 +315,41 @@ jobs:
...
+
+  check_annotations:
+    name: Check Annotations
+    needs:
+    - emscripten
+    - linux
+    - macos
+    - windows
+
+    if: github.event_name == 'pull_request'
```
Comment (TrueBrain) [inline .github/workflows/ci-build.yml]: maybe even `always() && github.event_name == 'pull_request'`, so it runs even if any of the `needs` jobs fail?

**Review by TrueBrain — APPROVED** [review]
I do not understand what the last step does, but I am sure it is fine :P

**Pull request merged**